db/migrate/20180823111741_move_location_led_state_to_asset_details_table.rb

```diff
@@ -0,0 +1,65 @@
+class MoveLocationLedStateToAssetDetailsTable < ActiveRecord::Migration[5.0]
+  class AssetDetail < ActiveRecord::Base
+    belongs_to :resource, :polymorphic => true
+
+    serialize :location_led_state
+  end
+
+  class PhysicalServer < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    has_one :asset_detail, :as => :resource, :dependent => :destroy
+  end
+
+  class PhysicalChassis < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    has_one :asset_detail, :as => :resource, :dependent => :destroy
+  end
+
+  def up
+    add_column :asset_details, :location_led_state, :string
+
+    PhysicalServer.includes(:asset_detail).each do |server|
+      asset_detail = AssetDetail.where('resource_id = ? and resource_type like ?', server.id, "%PhysicalServer")&.first
```

Comment:
Any reason not to use `server.asset_detail` here?

Reply:
This could be one-lined as:
```ruby
server.asset_detail&.update_attributes!(:location_led_state => server.location_led_state)
```

Reply:
Shouldn't we create the related `AssetDetail` if it doesn't exist so that we don't lose data?

Reply:
Actually I had some problems with `server.asset_detail` so I code this way, but now I did this change and everything looks fine, so thanks for the tip :+1: . About create the related `AssetDetail` I was concerned about those resources that, for some reason, did not have its `AssetDetail` filled, so I decide to respect that :smile: . I'm sure that Lenovo's resources always will have `AssetDetail` but maybe other vendor could not fill this information. Do you think that it is a valid concern :thinking: ?

Reply:
Oh, I think that I got you concern. If we don't create the relation, so the data will be loose. Therefore, agreed, we should create the relation